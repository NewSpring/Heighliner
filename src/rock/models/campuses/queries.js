
export default [
  "campuses(id: Int, name: String): [Campus]",
];
