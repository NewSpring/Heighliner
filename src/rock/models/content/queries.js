export default [
  `rockContent(
    channel: String!,
    limit: Int = 20,
    offset: Int = 0,
  ): [RockContent]`,
];
