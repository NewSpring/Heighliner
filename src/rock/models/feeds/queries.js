

export default [

  "userFeed(filters: [String]!): [Node]",

];
