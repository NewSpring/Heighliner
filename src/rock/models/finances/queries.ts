
export default [
  `savedPayments(limit: Int = 20, skip: Int = 0, cache: Boolean = true): [SavedPayment]`,

  `transactions(
    limit: Int = 20,
    skip: Int = 0,
    cache: Boolean = true,
    start: String,
    end: String,
    people: [Int],
  ): [Transaction]`,

  `scheduledTransactions(
    limit: Int = 20, skip: Int = 0, cache: Boolean = true, isActive: Boolean = true
  ): [ScheduledTransaction]`,

  `accounts(
    name: String, isActive: Boolean = true, isPublic: Boolean = true
  ): [FinancialAccount]`,

  `accountFromCashTag(cashTag: String!): FinancialAccount`,
];
