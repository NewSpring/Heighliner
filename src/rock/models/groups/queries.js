
export default [
  `groups(
    attributes: [String],
    limit: Int = 20,
    offset: Int = 0,
    query: String,
    campus: String,
    campuses: [String],
    clientIp: String,
  ): GroupSearch`,

  // XXX should this take a group type id?
  "groupAttributes: [DefinedValue]",
];
