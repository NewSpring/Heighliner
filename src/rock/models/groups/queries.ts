
export default [
  `groups(id: Int, name: String): [Group]`,
];
