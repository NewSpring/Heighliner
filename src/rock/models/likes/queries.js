export default [
  `recentlyLiked(
    limit: Int = 20,
    skip: Int = 0,
    cache: Boolean = true
  ): [Node]`
];
