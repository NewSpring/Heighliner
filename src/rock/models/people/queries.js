
export default [
  "people(email: String): [Person]",

  "person(guid: ID): Person",

  "currentPerson(cache: Boolean = false): Person",

  "currentFamily: [GroupMember]",
];
