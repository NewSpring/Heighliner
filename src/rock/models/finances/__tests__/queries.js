import queries from "../queries";

it("has all needed queries", () => {
  expect(queries).toMatchSnapshot();
});
