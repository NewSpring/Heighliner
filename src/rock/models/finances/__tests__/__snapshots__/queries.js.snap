// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has all needed queries 1`] = `
Array [
  "savedPayments(limit: Int = 20, skip: Int = 0, cache: Boolean = true): [SavedPayment]",
  "savedPayment(id: ID!): SavedPayment",
  "transactions(
    limit: Int = 20,
    skip: Int = 0,
    cache: Boolean = true,
    start: String,
    end: String,
    people: [Int] = [],
  ): [Transaction]",
  "scheduledTransactions(
    limit: Int = 20, skip: Int = 0, cache: Boolean = true, isActive: Boolean = true
  ): [ScheduledTransaction]",
  "accounts(
    allFunds: Boolean = false, name: String, isActive: Boolean = true, isPublic: Boolean = true, isTaxDeductable: Boolean = true
  ): [FinancialAccount]",
  "accountFromCashTag(cashTag: String!): FinancialAccount",
]
`;
